samples/grid/griddemo.cpp

```diff
@@ -1235,31 +1235,91 @@ void GridFrame::SetGridLineColour( wxCommandEvent& WXUNUSED(ev) )
     }
 }
 
+namespace
+{
+
+// Used when inserting/deleting rows/columns to allow changing by more than
+// one row/column at once. Note that the passed array is already sorted.
+bool IsConsecutiveSelection(const wxArrayInt& selection)
+{
+    const int selCount = static_cast<int>( selection.size() );
+
+    if ( !selCount )
+        return false;
+
+    const int first = selection[0];
+
+    for ( int i = 1; i < selCount; ++i )
+    {
+        if ( selection[i] != first + i )
+            return false;
+    }
+
+    return true;
+}
+
+} // anoymous namespace
 
 void GridFrame::InsertRow( wxCommandEvent& WXUNUSED(ev) )
 {
-    grid->InsertRows( grid->GetGridCursorRow(), 1 );
+    int pos, insertions;
+
+    const wxArrayInt selected = grid->GetSelectedRows();
+    if ( IsConsecutiveSelection(selected) )
+    {
+        pos = selected[0];
+        insertions = static_cast<int>( selected.size() );
+    }
+    else
+    {
+        pos = grid->GetGridCursorRow();
+        insertions = 1;
+    }
+
+    grid->InsertRows(pos, insertions);
 }
 
 
 void GridFrame::InsertCol( wxCommandEvent& WXUNUSED(ev) )
 {
-    grid->InsertCols( grid->GetGridCursorCol(), 1 );
+    int pos, insertions;
+
+    const wxArrayInt selected = grid->GetSelectedCols();
+    if ( IsConsecutiveSelection(selected) )
+    {
+        pos = selected[0];
+        insertions = static_cast<int>( selected.size() );
+    }
+    else
+    {
+         pos = grid->GetGridCursorCol();
+         insertions = 1;
+    }
+
+    grid->InsertCols(pos, insertions);
 }
 
 
 void GridFrame::DeleteSelectedRows( wxCommandEvent& WXUNUSED(ev) )
 {
-    if ( grid->IsSelection() )
+    if ( !grid->IsSelection() )
+        return;
+
+    wxGridUpdateLocker locker(grid);
+
+    const wxArrayInt selected = grid->GetSelectedRows();
+    if ( IsConsecutiveSelection(selected) )
     {
-        wxGridUpdateLocker locker(grid);
-        for ( int n = 0; n < grid->GetNumberRows(); )
-        {
-            if ( grid->IsInSelection( n , 0 ) )
-                grid->DeleteRows( n, 1 );
-            else
-                n++;
-        }
+        grid->DeleteRows( selected[0], static_cast<int>(selected.size()) );
+        return;
+    }
+
+    for ( int n = 0; n < grid->GetNumberRows(); )
+    {
+        if ( grid->IsInSelection( n , 0 ) )
```

Comment:
Just nitpicking...
```suggestion
        if ( grid->IsInSelection( n, 0 ) )
```

Reply:
I only moved this code around so far (it's original code which is now the fallback option for non-consecutive selections), which I still think doesn't justify any other editing.

Reply:
Oops, sorry, I hadn't realized this.